File: app/src/main/java/com/elvotra/clean/presentation/contract/PostDetailsContract.java
package com.elvotra.clean.presentation.contract;

import com.elvotra.clean.presentation.model.PostDetailsViewItem;

public interface PostDetailsContract {

    interface View extends BaseView<IPostDetailsPresenter> {

        void showPostDetails(PostDetailsViewItem postDetailsViewItem);

        void showNoResults();
    }

    interface IPostDetailsPresenter extends BasePresenter {

        void loadPost(int postId);
    }

}

File: app/src/main/java/com/elvotra/clean/presentation/contract/PostDetailsContract.kt
package com.elvotra.clean.presentation.contract

import com.elvotra.clean.presentation.model.PostDetailsViewItem

interface PostDetailsContract {

    interface View : BaseView<IPostDetailsPresenter> {

        val isActive: Boolean
        fun showPostDetails(postDetailsViewItem: PostDetailsViewItem)

        fun showNoResults()
    }

    interface IPostDetailsPresenter : BasePresenter {
        fun loadPost(postId: Int)
    }

}
